### ISSUE TITLE
Auto-populate event data from various sources?

### ISSUE THREAD

**Issue by rht** [fxfactorial/silicondzor#6]
- [ ] https://www.eventbrite.com/d/armenia--yerevan/events/?mode=search (e.g. the hour of code)
- [ ] https://www.meetup.com/cities/am/yerevan/
- [x] http://www.armtechcongress.com/
- [x] several FB groups / Twitter ann's
- [ ] http://10times.com/armenia/conferences

**Comment by fxfactorial** [comment]
Hi, good idea!

1) Eventbrite might be a good idea 
2) Meetup only has three groups for all of Armenia, one of which I made. 
3) arm tech seems to happen only once a year so that can be manually put in
4) Facebook is the key one

Of these only 1, 4 seem worth it to me to write any code, do you know if they have official APIs to pull down event data?

**Comment by rht** [comment]
> Of these only 1, 4 seem worth it to me to write any code, do you know if they have official APIs to pull down event data?

Hmm, there are indeed (for each in the list sans (3)). Tho you have to register the app to get the API tokens. (currently getting some tokens)

**Comment by rht** [comment]
1. https://www.eventbriteapi.com/v3/events/search/?token=0x5f3759df&location.latitude=40.1792&location.longitude=44.4991&location.within=50km (ref: https://www.eventbrite.com/developer/v3/endpoints/events/#ebapi-get-events-search), 4. https://developers.facebook.com/docs/graph-api/reference/v2.8/group/events

**Comment by fxfactorial** [comment]
@rht awesome! So there is a way, very nice that the group ID for the Facebook API looks pretty solid. We could have the server hit it up every 24 hours and update results into the DB.

**Comment by rht** [comment]
Yeah, (3) has a FB group with events, so nailing down (4) will do it for (3) as well.

**Comment by fxfactorial** [comment]
@rht I had to reorder some of your code, I believe that a token was needed for each request, as others some would crap out after a token was used. Also, I ran it manually to add some dates, and while the events did make it into the DB, they don't come up on the UI which makes me think that it probably an off by some date time amount. Can you check that it actually does populate the DB in such a way that is formatted correctly for the front end?

**Comment by rht** [comment]
Noted. A bit later then I will get back to this -- had to recompile nodejs7 in a while after some package updates.

**Comment by rht** [comment]
@fxfactorial here is an ecosystem of what has been implemented so far in the field of 'event aggregator': https://www.quora.com/What-are-some-good-conference-directories-or-calendars-other-than-Lanyrd, e.g. http://eventful.com/events?geo=city_id%3A91492 (fetched from eventbrite). The existing ones 1. aren't particularly covering the events granularly from each groups, 2. are proprietary hence can't be improved by anyone. However, there is also the concern with events that are public, and private events that can only be seen by the iterate app.

**Comment by fxfactorial** [comment]
@rht I'm still getting errors about things not being authenticated :( This FB wrapper SDK isn't very robust methinks. 

What do you mean about public/private events?

**Comment by rht** [comment]
What are the err messages? I was able to populate them all the way to the frontend.
Also there is a rate limit of 200 req/hr.

public/private: the former is events that are viewable without having to login to fb, the latter has a spectrum of shade from viewable to anyone with a fb token, to viewable to only the app.

**Comment by fxfactorial** [comment]
@rht basically the token doesn't seem to be used: 

```
 { message: 'A user access token is required to request this resource.',
Jan 12 18:34:42 debian-512mb-nyc3-01 node[9091]: type: 'OAuthException',
Jan 12 18:34:42 debian-512mb-nyc3-01 node[9091]: code: 102,
Jan 12 18:34:42 debian-512mb-nyc3-01 node[9091]: fbtrace_id: 'G7gMNc7wycl' } }
```

I'm trying out things out with the fb package.

Oh right right, well if things are a bit private then maybe we don't want them anyway...

**Comment by fxfactorial** [comment]
@rht Its pretty odd, I did a revert, then added everything other than the FB wrapper, but still get this

```
Jan 12 18:53:04 debian-512mb-nyc3-01 node[9679]: { message: 'A user access token is required to request this resource.',
Jan 12 18:53:04 debian-512mb-nyc3-01 node[9679]: type: 'OAuthException',
Jan 12 18:53:04 debian-512mb-nyc3-01 node[9679]: code: 102,
Jan 12 18:53:04 debian-512mb-nyc3-01 node[9679]: fbtrace_id: 'FafzM5THNOl' }
Jan 12 18:53:14 debian-512mb-nyc3-01 node[9679]: { message: 'A user access token is required to request this resource.',
Jan 12 18:53:14 debian-512mb-nyc3-01 node[9679]: type: 'OAuthException',
Jan 12 18:53:14 debian-512mb-nyc3-01 node[9679]: code: 102,
Jan 12 18:53:14 debian-512mb-nyc3-01 node[9679]: fbtrace_id: 'GZ84ssrdEpa' }
```

Its super odd and I initially thought maybe it was a event loop scheduling thing but its not...I think this `fb` is very fragile.

**Comment by rht** [comment]
I tested it when a `{access_token}` extra param wasn't put in.

**Comment by fxfactorial** [comment]
@rht yea, I tried that as well, will try again.

**Comment by fxfactorial** [comment]
@rht OMFG Now it does work

**Comment by rht** [comment]
I tested again the exact same code (w/ `setAccessToken`, w/ `{access_token}`):
```
[]
error occured when requesting a list of events for iterate
[ { description: 'Միջոցառման ժամանակ կծանոթանանք Ռումորս կազմակերպության կողմից ստեղծված Social Bridge և Rumors Monitoring հարթակների գործառույթներին։',
    end_time: '2017-01-19T19:00:00+0400',
    name: 'Ինչպե՞ս իրականացնել Facebook մոնիտորինգ',
    place: 
     { name: 'Innovative Solutions and Technologies Center - ISTC',
       location: [Object],
       id: '715035435194434' },
    start_time: '2017-01-19T18:00:00+0400',
    id: '1241963569223946' } ]
[ { description: 'Մայքրոսոֆթ ինովացիոն կենտրոն Հայաստանը և ArmSTQB-ն, ինչը հանդիսանում է ISTQB-ի Հայաստանյան ազգային խորհուրդը, սկսում են նոր դասընթաց նվիրված ISTQB հիմնական մակարդակին։ Դասընթացը անդրադառնում է թեստավորման գործընթացի հիմնական գործողություններին, թեստավորման նախագծման հիմնական եղանակներին, որոնք կուսուցանվեն տեսական դասընթացների և գործնական վարժությունների միջոցավ։ Դասընթացի ժամանակ կուսուցանվի թեստավորման մեթոդաբանությունը և անհատ թեստավորողների և թեստավորման ընդհանուր թիմի հետ կապված խնդիրները։  Դասընթացը նախապատրաստում է ISTQB հիմնական մակարդակի սետիֆիկացիային և հիմնված է International Software Testing Qualifications Board Foundation-ի կրթական ծրագրի վրա։\nՄեր դասախոս Արթուր Մկրտչյանը կսովորեցնի Ձեզ QA թեստավորման հիմունքները։',
...
```

**Comment by fxfactorial** [comment]
Yea, that's what I got as well. I wonder, why does it still mess up occasionally? The failure case seems indeterminate.

**Comment by fxfactorial** [comment]
@rht Up and running!

**Comment by rht** [comment]
((
after doing some minecraft-digging, I found a ton more list here https://classic.mapme.com/startup-armenia, e.g. for R&D
```
{ // yes there are comments -- this is not just a json
ANEL:ANEL-175466082623933, // no published events
ArmIndianCenter:261859337201788,
AybFoundation:282361768086,
BlockchainRnDHub:1858205724414486, // no published events
emgamelab:495404137307606, // no published events
techscidynamics:322065407957504,
YerevaNNlab: 1558194520864804 // rather than YerevanNN2
}
```

and for edu:
```
{
aiesecarmenia:248110690337,
AUArmenia:159219260801636,
armcodeacademy:420491148159763,
armroboticsarm:1441162332801281,
armscoop:169014669803270,
AybFoundation:282361768086,
AybSchool:1700171116883966, // no published events
CEEDArmenia: 214611175283547, // no public events
GITArmenia:447220742135729, // so this is what git stands for
gyumri.gtech:1470306789852783,
IPClub.am:1792943797602370,
LiveCodeYerevan:1237449446272856,
luseen.ltd:728174233968053,
LuysFoundation:225876370759572,
school.for.young.leaders:45596729524, // no published events
ShahumyanMediaLLC:108743009160204,
shushtech:580019628812316, // no published events
TumoCenter:193484930689681,
YerevanStateCollegeOfInformatics:242657529226272, // no published events
YoungProgrammersNGO:359657790800688
}
```

If some events are cross-shared several times, this info could be displayed as a share-count in a circle badge (this is one benefit of merging all these events into one db slab).
))

**Comment by fxfactorial** [comment]
Great! Pr it ! 

Also saw hye.life? 

(SD going on big refactor on another branch )
